src/DotNetWorker.Core/Context/Features/DefaultInputConversionFeature.cs

```diff
@@ -0,0 +1,118 @@
+            // string is not valid JSON. This manager will then continue with the next matching provider.
+            foreach (var converter in _inputConverterProvider.DefaultConverters)
+            {
+                var conversionResult = await ConvertAsyncUsingConverter(converter, converterContext);
+
+                if (conversionResult.IsSuccess)
+                {
+                    return conversionResult;
+                }
+            }
+
+            return default;
+        }
+                
+        private ValueTask<ConversionResult> ConvertAsyncUsingConverter(IInputConverter converter, ConverterContext context)
+        {
+            var conversionResultTask = converter.ConvertAsync(context);
+
+            if (conversionResultTask.IsCompletedSuccessfully)
+            {
+                return new ValueTask<ConversionResult>(conversionResultTask.Result);
+            }
+
+            return AwaitAndReturnConversionTaskResult(conversionResultTask);
+        }
+
+        private async ValueTask<ConversionResult> AwaitAndReturnConversionTaskResult(ValueTask<ConversionResult> conversionTask)
+        {
+            var result = await conversionTask;
+
+            return result;
+        }
+
+        /// <summary>
+        /// Gets an <see cref="IInputConverter"/> instance if converter context has information about what converter to be used.
+        /// </summary>
+        /// <param name="context">The converter context.</param>
+        /// <returns>An IInputConverter instance or null</returns>
+        private IInputConverter? GetConverterFromContext(ConverterContext context)
+        {
+            Type? converterType = default;
+
+            // Check a converter is specified on the conversionContext.Properties. If yes,use that.
+            if (context.Properties != null
+                && context.Properties.TryGetValue(PropertyBagKeys.ConverterType, out var converterTypeAssemblyQualifiedNameObj)
+                && converterTypeAssemblyQualifiedNameObj is string converterTypeAssemblyQualifiedName)
+            {
+                converterType = Type.GetType(converterTypeAssemblyQualifiedName);
+            }
+            else
+            {
+                // check the type used as "TargetType" has an "InputConverter" attribute decoration.
+                var converterAttribute = context.TargetType.GetCustomAttributes(_inputConverterAttributeType, inherit: true)
+                                                           .FirstOrDefault();
+
+                if (converterAttribute != null)
+                {
+                    converterType = ((InputConverterAttribute)converterAttribute).ConverterType;
+                }
+            }
+
+            if (converterType != null)
+            {
+                return this._inputConverterProvider.GetOrCreateConverterInstance(converterType);
+            }
+
+            return null;
```

Comment:
Just wondering why we return null here but default in other instances? (I'm assuming that default is the keyword and not a defined variable that I missed?)

Reply:
Yes, it is the `default` keyword. `return default` will also work here because `return default` will be lowered to `return null` in this case. I chose `return null` as I felt that is more readable in the case of this method.

I updated the other method which was doing `return default` so that it is more readable what it returns.